packages/form-core/src/FieldApi.ts

```diff
@@ -1387,6 +1397,14 @@ export class FieldApi<
                 // Prefer the error message from the field validators if they exist
                 error ? error : errorFromForm[errorMapKey],
             },
+            errorSourceMap: {
+              ...prev.errorSourceMap,
+              [getErrorMapKey(validateObj.cause)]: error
```

Comment:
Can we do that here and a couple of lines above?

```suggestion
              [errorMapKey]: error
```

Reply:
Nice! Can't believe I didn't see that! Updated.